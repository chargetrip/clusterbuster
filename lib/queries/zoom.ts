/**
 * @description The dafault implementation of zoom to distance
 */
export const defaultZoomToDistance = (zoomLevel: number, radius: number = 15) =>
  radius / Math.pow(2, zoomLevel);
