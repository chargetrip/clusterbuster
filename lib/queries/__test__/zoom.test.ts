import { defaultZoomToDistance } from '../zoom';

describe('defaultZoomToDistance', () => {
  it('should work', () => {
    expect(defaultZoomToDistance(0, 2)).toMatchSnapshot();
    expect(defaultZoomToDistance(1, 2)).toMatchSnapshot();
    expect(defaultZoomToDistance(5, 2)).toMatchSnapshot();
    expect(defaultZoomToDistance(10, 2)).toMatchSnapshot();
    expect(defaultZoomToDistance(14, 2)).toMatchSnapshot();
    expect(defaultZoomToDistance(0)).toMatchSnapshot();
    expect(defaultZoomToDistance(7)).toMatchSnapshot();
    expect(defaultZoomToDistance(14)).toMatchSnapshot();
  });
});
