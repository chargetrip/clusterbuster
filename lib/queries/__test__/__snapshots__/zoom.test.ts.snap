// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultZoomToDistance should work 1`] = `2`;

exports[`defaultZoomToDistance should work 2`] = `1`;

exports[`defaultZoomToDistance should work 3`] = `0.0625`;

exports[`defaultZoomToDistance should work 4`] = `0.001953125`;

exports[`defaultZoomToDistance should work 5`] = `0.0001220703125`;

exports[`defaultZoomToDistance should work 6`] = `15`;

exports[`defaultZoomToDistance should work 7`] = `0.1171875`;

exports[`defaultZoomToDistance should work 8`] = `0.00091552734375`;
